lib/Infer/ExhaustiveSynthesis.cpp

```diff
@@ -731,6 +758,16 @@ ExhaustiveSynthesis::synthesize(SMTLIBSolver *SMTSolver,
     separatePCs(PCs, PCsCopy, InstCache, BlockCache, IC, &ConstMap, false);
 
     InstMapping Mapping2(LHS, I2);
+
+    if (LSBPruning && !GuessHasConstant) {
```

Comment:
this idea should still work fine when we have synthesized constants on the RHS, right?

Reply:
Yes, let's do that with another PR, added a TODO.

Reply:
actually I don't think it's as easy as I thought-- so definitely save for later